File: qc-application/src/main/java/com/eager/questioncloud/application/message/FailChargePointPaymentMessage.java
package com.eager.questioncloud.application.message;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class FailChargePointPaymentMessage {
    private String paymentId;
}

File: qc-application/src/main/java/com/eager/questioncloud/application/message/FailChargePointPaymentMessage.kt
package com.eager.questioncloud.application.message;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class FailChargePointPaymentMessage {
    private int failCount;
    private String paymentId;

    public static FailChargePointPaymentMessage create(String paymentId) {
        return new FailChargePointPaymentMessage(0, paymentId);
    }

    public void increaseFailCount() {
        this.failCount++;
    }
}
